metaseq/modules/multihead_attention.py

```diff
@@ -385,7 +431,20 @@ def forward(
             # the transpose is a no-op copy before view, thus unnecessary
             attn = attn.contiguous().view(tgt_len, bsz, embed_dim)
         else:
-            attn = attn.transpose(0, 1).contiguous().view(tgt_len, bsz, embed_dim)
+            attn = attn.transpose(0, 1).contiguous()
+            if isinstance(attn, ShardedTensor):
+                sharding_spec = attn.sharding_spec()
+                sharding_spec.dim = -1
+                st_size = (
+                    attn.size(0),
+                    attn.size(1) // world_size,
+                    attn.size(2) * world_size,
+                )
+                attn = ShardedTensor._init_from_local_tensor(
+                    attn.local_tensor(), sharding_spec, st_size
+                )
+            attn = attn.view(tgt_len, bsz, embed_dim)
```

Comment:
Hmm so why can't our `view` op directly deal with this?

Reply:
Because View does not change sharding spec automatically.